test/authorizer.int.ts

```diff
@@ -0,0 +1,137 @@
+import { SimulatedAuthorizer } from "../src/core/auth/provider/simulated/simulated_authorizer";
+
+require('dotenv').config()
+
+describe('Authorizer (Hosted and Simulated)', () => {
+
+    it('Should give HostedAuthorizer for configuration', () => {
+        // Arrange and act
+        const authorizer = Core.getAuthorizer({ provider: "Hosted" });
+        // Assert
+        expect(authorizer).toBeInstanceOf(HostedAuthorizer);
+    });
+
+    it('Should give SimulatedAuthorizer for configuration', () => {
+        // Arrange and act
+        const authorizer = Core.getAuthorizer({ provider: "Simulated" });
+        // Assert
+        expect(authorizer).toBeInstanceOf(SimulatedAuthorizer);
+    });
+
+    it('Should pick up host url from env if not given', () => {
+        // Arrange
+        const envUrl = "sample-env-url";
+        process.env.AUTHURL = envUrl;
+        // Act
+        const authorizer = Core.getAuthorizer({ provider: "Hosted" }) as HostedAuthorizer;
+        // Assert
+        expect(authorizer.baseUrl).toEqual(envUrl);
+
+    });
+    it('Should configure based on the url provided', () => {
+        // Arrange
+        const envUrl = process.env.AUTHURL as string;
+        // Act
+        const authorizer = Core.getAuthorizer({ provider: "Hosted", apiUrl: "http://sample.com/api" }) as HostedAuthorizer;
+        // Assert
+        expect(authorizer.baseUrl).not.toEqual(envUrl);
+    });
+
+    it('Should give error for no permissions', async () => {
+        // Arrange
+        const authorizer = Core.getAuthorizer({ provider: "Hosted" });
+        const permission = new PermissionRequest({
+            userId: "abc",
+            orgId: "sdfs",
+            permssions: [],
+            shouldSatisfyAll: false
+        });
+        // Act and Assert
+        await expect(authorizer?.hasPermission(permission)).rejects.toEqual("No roles provided");
+    });
+
+    it('Should give simulated result as false when userId is as per predefined', async () => {
+        // Arrange
+        const authorizer = Core.getAuthorizer({ provider: "Simulated" });
+        const permission = new PermissionRequest({
+            userId: "000000-0000-0000-0000-000000",
+            orgId: "sdfs",
+            permssions: [],
+            shouldSatisfyAll: false
+        });
+        // Act and Assert
+        await expect(authorizer?.hasPermission(permission)).resolves.toEqual(false);
+
+    });
+
+    it('Should give simulated result as expected (true)', async () => {
+        // Arrange
+        const authorizer = Core.getAuthorizer({ provider: "Simulated" });
+        const permission = new PermissionRequest({
+            userId: "abc",
+            orgId: "sdfs",
+            permssions: [],
+            shouldSatisfyAll: true
+        });
+        // Act and assert
+        await expect(authorizer?.hasPermission(permission)).resolves.toEqual(true);
+    });
+
+    it('Should respond true when user has permission', async () => {
```

Comment:
for all Integration test create section (describe) or create file .integration.ts and move all tests